src/Libplanet.Net/Swarm.BlockCandidate.cs

```diff
@@ -179,15 +179,15 @@ private BlockChain AppendPreviousBlocks(
                 throw new InvalidGenesisBlockException(
                     msg,
                     branchpoint.Hash,
-                    workspace.Genesis.Hash);
+                    blockChain.Genesis.Hash);
             }
             else
             {
                 _logger.Debug(
                     "Trying to fork... at {MethodName}()",
                     nameof(AppendPreviousBlocks)
                 );
-                workspace = workspace.Fork(branchpoint.Hash);
```

Comment:
Using BlockChain instead of workspace means that there is no point in forking here. To put it another way, in my opinion, the need for Fork is a situation where it is assumed that Reorg still exists. What do you think about removing Fork first?

Reply:
The problem is that it is not guaranteed by the current API that forking would not occur. With PBFT, the whole `else` block would not be run in the first place, but it is left as it is to reduce the scope of this PR which mainly focuses on the internal logic of `GetDemandBlockHashes()`. Cleaning up `Fork`ing, other unnecessary conditional branching, reimplementation of `FindBranchPoint`, etc. to name the few will be done gradually in follow-up PRs. 😅